lib/rules/max-top-level-suites.js

```diff
@@ -0,0 +1,49 @@
+'use strict';
+
+/**
+ * @fileoverview Limit the number of top-level suites in a single file
+ * @author Alexander Afanasyev
+ */
+
+var R = require('ramda'),
+    astUtil = require('../util/ast'),
+    defaultSuiteLimit = 1;
+
+module.exports = function (context) {
+    var stack = [],
+        topLevelDescribes = [],
+        suiteLimit = context.options[0];
+
+    if (R.isNil(suiteLimit)) {
+        suiteLimit = defaultSuiteLimit;
+    } else {
+        suiteLimit = suiteLimit.limit;
```

Comment:
> `TypeError: Cannot read property 'loc' of undefined`

No, that sounds like a normal error.

`suiteLimit = defaultSuiteLimit ? R.isNil(options) : options.limit`
`defaultSuiteLimit` is a truthy value, so `suiteLimit` will evaluate to `R.isNil(options)`, meaning `true` or `false`.

When you then compare the number of top-level suites to this boolean, it will evaluate to true if there is at least one. Then, you report an error, but pass undefined as the node (as `topLevelDescribes[true]` doesn't exist), which ESLint then uses to find `loc`. Crash!

tl;dr: You should have written it as `suiteLimit = R.isNil(options.limit) ? defaultSuiteLimit : options.limit)`. I would have done it the way I proposed, but your way should work fine too ;)